src/drivers/uavcan_v1/Subscribers/Gnss.hpp

```diff
@@ -0,0 +1,99 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file Gnss.hpp
+ *
+ * Defines basic functionality of UAVCAN v1 GNSS subscription
+ *
+ * @author Jacob Crabill <jacob@flyvoly.com>
+ */
+
+#pragma once
+
+// DS-15 Specification Messages
+#include <reg/drone/physics/kinematics/geodetic/Point_0_1.h>
+
+#include "Subscriber.hpp"
+
+class UavcanGnssSubscription : public UavcanSubscription
+{
+public:
+	UavcanGnssSubscription(CanardInstance &ins, UavcanParamManager &pmgr, const char *uavcan_pname) :
+		UavcanSubscription(ins, pmgr, uavcan_pname) { };
+
+	void subscribe() override
+	{
+		// Subscribe to messages reg.drone.physics.kinematics.geodetic.Point.0.1
+		canardRxSubscribe(&_canard_instance,
+				  CanardTransferKindMessage,
+				  _port_id,
+				  reg_drone_physics_kinematics_geodetic_Point_0_1_EXTENT_BYTES_,
+				  CANARD_DEFAULT_TRANSFER_ID_TIMEOUT_USEC,
+				  &_canard_sub);
+
+		/** TODO: Add additional GPS-data messages: (reg.drone.service.gnss._.0.1.uavcan):
```

Comment:
Hey @JacobCrabill a question about terminology here, does the UavcanSubscription consists of multiple subscriptions from a reg.drone.service e.g.  https://github.com/UAVCAN/public_regulated_data_types/blob/master/reg/drone/service/gnss/_.0.1.uavcan or just a single subscription to a specific topic e.g. point_kinematics.

Reply:
I haven't come up with a good answer yet.  Right now it's meant for just 1 topic, but for services consisting of multiple topics, I haven't come up with a good solution yet.  It seems overly bulky to have to create a Subscription for every sub-topic of a service, and then the data wouldn't be consolidated in one area either.  Considering making a "SubscriptionMulti" class.